src/main/scala/com/yotpo/metorikku/metric/Metric.scala

```diff
@@ -117,28 +133,52 @@ case class Metric(configuration: Configuration, metricDir: File, metricName: Str
     }
   }
 
+  private def getEpochLagTime(dataFrame: DataFrame, lagReporterTimeColumn: Option[String]): Option[Long] ={
+    lagReporterTimeColumn match {
+      case Some(timeColumn) => {
+        try {
+          val maxLagValue = dataFrame.agg({timeColumn.toString -> "max"}).collect()(0)(0)
+          if (maxLagValue.isInstanceOf[java.sql.Timestamp]) {
+            return Option(dataFrame.agg({timeColumn.toString -> "max"}).collect()(0).getTimestamp(0).getTime())
```

Comment:
You're running it again? why? you can check the schema beforehand and figure out the type

Reply:
Also I think you need to cache before... as this will basically calculate everything